mindmeld/auto_annotator.py

```diff
@@ -835,5 +839,300 @@ def valid_entity_check(self, entity):
+                    if not (
+                        candidate["dim"] == "amount-of-money"
+                        and candidate["value"]["unit"] == "unknown"
+                    ):
+                        selected_candidates.append(candidate)
+                        break
+        return selected_candidates
+
+    @staticmethod
+    def convert_duckling_candidates(duckling_candidates):
+        """Converts duckling candidates into entity dictionaries.
+        Args:
+            duckling_candidates (list): List of Duckling candidates
+        Returns:
+            entities (list): List of converted entities.
+        """
+        return [
+            {
+                "body": entity["body"],
+                "start": entity["start"],
+                "end": entity["end"],
+                "dim": entity["entity_type"],
+                "value": entity["value"],
+            }
+            for entity in duckling_candidates
+        ]
+
+    @property
+    def supported_entity_types(self):  # pylint: disable=W0236
+        """
+        Returns:
+            supported_entity_types (list): List of supported entity types.
+        """
+        return SPACY_ANNOTATOR_SUPPORTED_ENTITIES
+
+    @staticmethod
+    def _has_overlap(span_one, span_two):
+        """Determines whether two spans overlap. Spans are two-element tuples with the format:
+        (start_index, end_index + 1).
+        """
+        return span_one[1] > span_two[0] and span_two[1] > span_one[0]
+
+    @staticmethod
+    def _is_larger(span_one, span_two):
+        """Determines whether the first spans is larger than the second. Spans are two-element
+        tuples with the format: (start_index, end_index + 1).
+        """
+        return span_one[1] - span_two[0] > span_one[1] - span_two[0]
+
+    @staticmethod
+    def _filter_mid_token_spans(spans, sentence):
+        """Removes any spans that end in-between a token.
+
+        Args:
+            spans (list): List of tuples representing candidate spans (start_index, end_index + 1).
+            sentence (str): The original sentence parsed by duckling.
+        Returns:
+            filtered_spans (list): List of spans that do not end mid-token.
+        """
+        space_indices = [i for i in range(len(sentence)) if sentence[i] == " "]
+        return [
+            span
+            for span in spans
+            if span[1] in space_indices or span[1] >= len(sentence) - 2
+        ]
+
+    @staticmethod
+    def _get_largest_non_overlapping_candidates(spans):
+        """Finds the set of the largest non-overlapping candidates.
+
+        Args:
+            spans (list): List of tuples representing candidate spans (start_index, end_index + 1).
+        """
+        selected_spans = []
+        for span in spans:
+            no_overlaps = True
+            for i, selected_span in enumerate(selected_spans):
+                if MultiLingualAnnotator._has_overlap(span, selected_span):
```

Comment:
```
no_overlaps = MultiLingualAnnotator._has_overlap(span, selected_span)
if not no_overlaps:
    continue
if MultiLingualAnnotator._has_overlap(span, selected_span):
    selected_spans[i] = span
```

Reply:
Better indentation ^

Reply:
Refactored this function, using cleaner logic:

```
        for span in spans:
            has_overlaps = [
                MultiLingualAnnotator._has_overlap(span, selected_span)
                for selected_span in selected_spans
            ]
            if not any(has_overlaps):
                selected_spans.append(span)
```